src/app/drive/services/worker.service/downloadWorkerHandler.ts

```
@@ -70,65 +69,40 @@ export class DownloadWorkerHandler {
     const { result } = messageData;
     let aborted = false;
 
+    const abortWriter = async () => {
+      if (aborted) return;
+
+      try {
+        worker.postMessage({ type: 'abort' });
+        await writer.abort();
+        aborted = true;
+      } catch {
+        // NO OP
+      } finally {
+        worker.terminate();
+        reject('Aborted');
```

A promise reject should return an error, let's keep the correctness